pkg/querier/queryrange/query_range.go

```diff
@@ -205,6 +207,14 @@ func (prometheusCodec) DecodeRequest(_ context.Context, r *http.Request) (Reques
 
 	result.Query = r.FormValue("query")
 	result.Path = r.URL.Path
+
+	for _, value := range r.Header.Values(cacheControlHeader) {
+		if value == noStoreValue {
```

Comment:
You should use the `string.Contains()` because of the reason explain in the previous comment. Iterating on `r.HeaderValues()` doesn't cover the case the header value is something like `no-cache,no-store`.

Reply:
Okay, sorry I missed that. Umm, what if the request sets `Cache-Control=no-cache` and `Cache-Control=no-store` at the same time?  `r.Header.Get()` only returns the first value.

Reply:
I think I got the answer. Just want to double check. Seems it is not necessary and multi-value header is only valid in comma-separated format. Let me use `string.Contains()`. Thanks!

![Screenshot from 2020-08-28 12-51-56](https://user-images.githubusercontent.com/25150124/91592953-4f369a00-e92d-11ea-8881-edeea9734396.png)

Reply:
Sorry, I missed that! Thanks for pointing this out!